### PR TITLE
Newbranch

### PR DESCRIPTION
**Pull request by Kubik-Wrubik**

### CODE REVIEW

File: src/Island/Entities/Animal.java [hunk 1]
```diff
@@ -0,0 +1,213 @@
+package Island.Entities;
```
Comment (Volodymyr95) [inline src/Island/Entities/Animal.java]: package name should start from lower case letter

File: src/Island/Entities/Animal.java [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public abstract class Animal implements Runnable, Entity{
+	private static final AtomicInteger ID = new AtomicInteger(1);
+	private final Object lock;
+
+	public final String name;
```
Comment (Volodymyr95) [inline src/Island/Entities/Animal.java]: why all fields are public?

File: src/Island/Entities/Animal.java [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+		}
+	}
+
+	/**
+	 * the method selects a valid direction of animal movement(up, down, left, right)
+	 */
+	public void move(){
+		final Island island = currentLocation.getIsland();
+
+		int directionsAmount = 4;
```
Comment (Volodymyr95) [inline src/Island/Entities/Animal.java]: 4 is magic number

File: src/Island/Entities/Animal.java [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+		List<Directions> directionsList = new ArrayList<>(Arrays.asList(Directions.UP, Directions.RIGHT, Directions.DOWN, Directions.LEFT));
+
+		if(currentLocation.positionX == 0){
```
Comment (Volodymyr95) [inline src/Island/Entities/Animal.java]: 0 is magic number

File: src/Island/Entities/Entity.java [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package Island.Entities;
+
+/**
+ * Marker interface for {@link Animal}, {@link Meat} and {@link Vegetation}
+ */
+public interface Entity {}
```
Comment (Volodymyr95) [inline src/Island/Entities/Entity.java]: why do u need it?

File: src/Island/Entities/Names.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
+package Island.Entities;
+
+/**
+ * enum of names for born animals
+ */
+public enum Names {
```
Comment (Volodymyr95) [inline src/Island/Entities/Names.java]: why do u need this enum?

File: src/Island/Entities/Predators/Wolf.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import Island.Entities.Animal;
+import Island.Entities.Herbivorous.*;
+
+public class Wolf extends Predator {
+	public Wolf(String name, Cell cell, Object lock){
+		super(name, 50, 3, 8, 14, 2, cell, lock);
+	}
+
+	public int getPrey(Animal animal){
+		if(animal instanceof Horse) return 10;
```
Comment (Volodymyr95) [inline src/Island/Entities/Predators/Wolf.java]: u can make a make where key is animal and value is number

File: src/Island/Entities/Animal.java (hunk 4 above)
Comment (Kubik-Wrubik) [reply src/Island/Entities/Animal.java]: here 0 is the edge of island from left. How can I replace this magic number?

**Pull request merged**